libs/contrib/Data/Always.idr

```diff
@@ -0,0 +1,39 @@
+module Data.Always
+
+import Data.Elem
+
+%default total
+
+data Always : (x -> Type) -> List x -> Type where
+  Nil : Always f []
+  (::) : f ty -> Always f l -> Always f (ty :: l)
+
+index : Elem x xs -> Always f xs -> f x
+index Here (x :: _) = x
+index (There elem) (_ :: xs) = index elem xs
+
+replaceAt : Elem x xs -> f x -> Always f xs -> Always f xs
+replaceAt Here element (_ :: xs) = element :: xs
+replaceAt (There n) element (x :: xs) = x :: replaceAt n element xs
+
+replaceAndIndex : {elements:Always f xs} -> index elem (replaceAt elem x elements) = x
+replaceAndIndex {elem = Here} {elements = Nil} impossible
+replaceAndIndex {elem = Here} {elements = (_ :: _)} = Refl
+replaceAndIndex {elem = There elem} {elements = Nil} impossible
+replaceAndIndex {elem = There elem} {elements = (_ :: _)} = replaceAndIndex
+
+replaceAndIndexElsewhere : {elements:Always f xs} -> (elem1 = elem2 -> Void) -> index elem1 (replaceAt elem2 x elements) = index elem1 elements
```

Comment:
See above - there should be spaces around the colon, and it should probably be an `auto` implicit.